toolchains/private/small_set.bzl

```diff
@@ -0,0 +1,34 @@
+"""Semantically equivalent to a depset, but optimized for tiny sizes and for iteration (depset requires .to_list each time you want to iterate)."""
+
+visibility("//toolchains")
+
+def create_small_set(values):
+    """Creates a small set from a Sequence of values.
+
+    Args:
+        values: (Sequence[provider]) Values to merge
+
+    Returns:
+        A deduplicated set small set of providers.
```

Comment:
```suggestion
        A deduplicated small set of providers.
```

Reply:
Isn't the returned value just a deduplicated collection? It doesn't support efficient `in` for example.

Reply:
That's correct - it's not optimized for `in` (and we never use it for in).

It's optimized for use cases where we create it once in the depgraph, it remains small, and then on every single action we need to call `.to_list()`. This essentially just stores the `.to_list()` version of it